.github/workflows/CD.yml

```diff
@@ -1,8 +1,11 @@
 name: CD
+
 on:
+  workflow_call:
+  pull_request:
```

Comment:
This line should not be here - you are wanting to deploy to your fly io site whatever is on main but with this setting you're deploying the latest PR up there, e.g. if I made a new PR which was an empty Next app, that would overwrite whatever is deployed

Reply:
@j4ck-wright I see what you are saying.

The course says this: "Remember that we want only to bump the version when the change happens to the main branch!"

But as you have said, for this to happen the CI has already run so it is pointless running the jobs again. 

I have added code that I think tags on merge to the main. It will only bump the minor number. 

I didn't want to do this originally because there us no way of testing it will tag unless I merge to main.